projects/wave-core-new/src/lib/layer-collections/layer-collection-list/layer-collection-list.component.ts

```diff
@@ -0,0 +1,262 @@
+import {CdkVirtualScrollViewport} from '@angular/cdk/scrolling';
+import {filter, mergeMap, scan, tap} from 'rxjs/operators';
+import {LayoutService} from '../../layout.service';
+import {
+    GeoEngineError,
+    LayerCollectionItem,
+    LayerCollectionLayer,
+    RasterResultDescriptorDict,
+    UUID,
+    VectorResultDescriptorDict,
+} from '../../backend/backend.model';
+import {LayerCollectionService} from '../layer-collection.service';
+import {createIconDataUrl} from '../../util/icons';
+import {ProjectService} from '../../project/project.service';
+import {NotificationService} from '../../notification.service';
+import {RasterLayer, VectorLayer} from '../../layers/layer.model';
+import {
+    ClusteredPointSymbology,
+    LineSymbology,
+    PointSymbology,
+    PolygonSymbology,
+    RasterSymbology,
+    VectorSymbology,
+} from '../../layers/symbology/symbology.model';
+import {colorToDict} from '../../colors/color';
+import {RandomColorService} from '../../util/services/random-color.service';
+
+@Component({
+    selector: 'wave-layer-collection-list',
+    templateUrl: './layer-collection-list.component.html',
+    styleUrls: ['./layer-collection-list.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LayerCollectionListComponent implements OnInit, AfterViewInit {
+    @ViewChild(CdkVirtualScrollViewport)
+    viewport!: CdkVirtualScrollViewport;
+
+    @Input() collection?: UUID = undefined;
+
+    readonly loadingSpinnerDiameterPx: number = 3 * LayoutService.remInPx;
+
+    source?: LayerCollectionItemDataSource;
+
+    constructor(
+        private readonly layerService: LayerCollectionService,
+        private readonly layoutService: LayoutService,
+        private readonly projectService: ProjectService,
+        private readonly notificationService: NotificationService,
+        private readonly randomColorService: RandomColorService,
+    ) {}
+
+    ngOnInit(): void {
+        this.source = new LayerCollectionItemDataSource(this.layerService, this.collection);
+    }
+
+    ngAfterViewInit(): void {}
+
+    /**
+     * Fetch new data when scrolled to the end of the list.
+     */
+    onScrolledIndexChange(_scrolledIndex: number): void {
+        const end = this.viewport.getRenderedRange().end;
+        const total = this.viewport.getDataLength();
+
+        // only fetch when scrolled to the end
+        if (end >= total) {
+            this.source?.fetchMoreData();
+        }
+    }
+
+    trackById(_index: number, item: LayerCollectionItem): UUID {
+        return item.id;
+    }
+
+    icon(item: LayerCollectionItem): string {
+        return createIconDataUrl(item.type);
+    }
+
+    select(item: LayerCollectionItem): void {
+        if (item.type === 'collection') {
+            this.layoutService.setSidenavContentComponent({
+                component: LayerCollectionListComponent,
+                config: {collection: item.id},
+                keepParent: true,
```

Comment:
You should remove this, as it de-highlights the current tab.

Instead, you could add a simple "<" | ">" menu where you can go back and forth.
You don't need to focus on the old spot in the list just yet.

Reply:
so how would I go about this? Add a new `LayerCollectionNavigationComponent` that provides the buttons and a div where the actual list is shown and then an Array of Collection Ids that keep track on the navigation? Or is there an existing Angular component that can help me with this?

Reply:
Yeah, just put the  "<" | ">" buttons on top and the content below. I guess the thing with the arrays should work.

For the content, you can use a [portal](https://material.angular.io/cdk/portal/overview) to put the listing itself into.

Reply:
I managed to get the portal working after all: cccc2b7d949e49c7ef04ff378a717339e5d9af26